### PR TITLE
Add support for YAML Steps Pipeline

### LINKED ISSUE #61: Add support for YAML Pipeline
https://buildkite.com/changelog/99-introducing-the-yaml-steps-editor

After migrating to YAML steps, [create_pipeline()](https://github.com/pyasi/pybuildkite/blob/master/pybuildkite/pipelines.py#L45) no longer works.

Buildkite throws an HTTP 500 error when trying to use
```
buildkite.pipelines().create_pipeline(
    organization = "org",
    pipeline_name = "pipeline",
    git_repository = "git@github.com:pyasi/pybuildkite.git"
)
```

**Error**
```
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://api.buildkite.com/v2/organizations/dropbox/pipelines/?per_page=100
```

The same error can be reproduced using `curl`.

In order to create a pipeline, we must use the new param `configuration`  in https://buildkite.com/docs/apis/rest-api/pipelines#create-a-pipeline-yaml-pipelines

### PR DESCRIPTION
**Pull request by changusmc**
fixes #61

### CODE REVIEW

**Comment by pyasi** [comment]
Nice Find! If the create_pipeline method is throwing a 500, id recommend we remove it altogether. Thoughts?

File: pybuildkite/pipelines.py [hunk 1]
```diff
@@ -75,6 +75,28 @@ def create_pipeline(
 
         return self.client.post(self.path.format(organization), body=data)
 
+    def create_yaml_pipeline(
```
Comment (pyasi) [inline pybuildkite/pipelines.py:78]: To add a bit more detail, perhaps we formerly deprecate the previous call and log out that it will 500.

Comment (changusmc) [reply pybuildkite/pipelines.py:78]: One thing I forgot to mention is the old call (very likely) still works if the user hasn't migrated to YAML pipelines.

Comment (changusmc) [reply pybuildkite/pipelines.py:78]: One thing I forgot to mention is the old call (very likely) still works if the user hasn't migrated to YAML pipelines.

**Comment by changusmc** [comment]
> Nice Find! If the create_pipeline method is throwing a 500, id recommend we remove it altogether. Thoughts?

Based on the Buildkite REST API docs, it seems the old method still works they didn't mention that when users migrate to the YAML pipeline, you cannot use the `steps` in the body.

> You can use this endpoint to create both YAML pipelines and visual step pipelines.
https://buildkite.com/docs/apis/rest-api/pipelines#create-a-pipeline

File: pybuildkite/pipelines.py (hunk 1 above)
Comment (pyasi) [reply pybuildkite/pipelines.py:78]: Got it, that makes sense. In that case can we add specific documentation on this and the other method- perhaps referencing Buildkite's docs about the migration?

Comment (changusmc) [reply pybuildkite/pipelines.py:79]: Added the docs in https://github.com/pyasi/pybuildkite/pull/62/commits/445c4028d15e3ae33fed4fe642d3807364f4a052

File: pybuildkite/pipelines.py [hunk 2]
```diff
@@ -98,6 +122,7 @@ def update_pipeline(
         name: str = None,
         provider_settings: dict = None,
         repository: str = None,
+        configuration: str = None,
         steps: dict = None,
```
Comment (pyasi) [inline pybuildkite/pipelines.py:126]: Do we need to catch and throw an error if somebody tried to use both of these arguments? We could throw a custom Exception and explain which one they should use.

Comment (changusmc) [reply pybuildkite/pipelines.py:126]: Good call, let me add that

**Comment by pyasi** [comment]
LGTM! Thanks for the contributions!

**Comment by changusmc** [comment]
> LGTM! Thanks for the contributions!

np, it was very easy to contribute. How often do you publish the package?

**Comment by pyasi** [comment]
Published this one already: https://github.com/pyasi/pybuildkite/releases/tag/v1.2.0

**Pull request merged**